### ISSUE TITLE
export of initWebReaderSW from sw.ts

### ISSUE THREAD

**Issue by AntonOfTheWoods** [NYPL/web-reader#113]
I was looking to complete the integration of `web-reader` (finally!) and was getting on to `initWebReaderSW`. It looks like it isn't exported though, so can you confirm that needs to happen before I can follow the docs for setting up pre-caching? If so, I'll get onto that with the other PR (https://github.com/NYPL-Simplified/web-reader/issues/107) I owe you!

**Comment by samanthaandrews** [comment]
Ok so I've done a little digging and found that this is how we are importing that `initWebReaderSW` function in Circulation Patron Web: https://github.com/NYPL-Simplified/circulation-patron-web/blob/qa/src/serviceworker/index.ts

You're right about it not being exported. I'll defer to @kristojorg on whether we should change this, as he has slightly more context on the background of this package. He's out this week but I'll make sure to follow up with him when he returns.

**Comment by AntonOfTheWoods** [comment]
Thanks again for your reply @samanthaandrews . Actually it looks like importing ths way might avoid the `document` issues that I referred to in my other issue https://github.com/NYPL-Simplified/web-reader/issues/114 !

That said, it's not super clear to me how things are supposed to get served from the cache, if at the bottom of https://github.com/NYPL-Simplified/web-reader/blob/main/src/ServiceWorker/sw.ts the code which looks like it does that is commented out. I couldn't see how this is done in `circulation-patron-web`. I probably don't understand everything but the caching seems to get done by `swr` (via the `mutate` method, which has a `cache.set`) but how this relates to the commented out code, I can't really tell.

Or is the idea that users of this functionality will copy/paste the commented out code in their own service worker and call this *after* calling `initWebReaderSW`?

**Comment by samanthaandrews** [comment]
After reading your message I remembered that we actually disabled the publication service worker in this PR that was merged on Nov 22: https://github.com/NYPL-Simplified/circulation-patron-web/pull/309/files

That code might help. However, it seems that we are aware that we need to revisit our service worker caching strategy in order to get this working properly. We have an open issue in Jira to track this, which has the following description:

> The web-reader Service Worker is supposed to prefetch and cache book assets for offline viewing. It should only do this during network idle, so that prefetching doesn't block other requests. However, it currently does this on book open, and it appears all the resources are fetched with equal priority. Thus book loading takes a looooong time. To minimize this, we should consider other methods of prefetching including <link prefetch> style, manually setting a low priority on SW prefetch calls, or only sending those calls during network idle.

**Comment by AntonOfTheWoods** [comment]
Thanks for your reply. Is this your main production use at the moment? Otherwise, I can't really see what I was doing wrong (apart from not understanding how to import properly...).

Actually I have decided that I will just import all of the constituent parts and have my own version of `initWebReaderSW`. Given how central the SW is in my architecture, I can't really let anything else do stuff I don't totally control.

Given the problems you talk about with loading taking a loooong time, can I assume you are talking about massive picture books on bad mobile connections? For me, with books up to several MB being sent from a $10 per month VPS in another continent with horrible network, they are cached before the reader fully loads!
